modules/xmpp/SDPUtil.js

```diff
@@ -443,6 +443,30 @@ var SDPUtil = {
             .split(" ")
             .map(ssrcStr => parseInt(ssrcStr));
     },
+
+    /**
+     * Sets the given codecName as the preferred codec by
+     *  moving it to the beginning of the payload types
+     *  list (modifies the given mline in place)
+     * @param {object} videoMLine the video mline object from
+     *  an sdp as parsed by transform.parse
+     * @param {string} the name of the preferred codec
+     */
+    preferVideoCodec: function(videoMLine, codecName) {
+        let payloadType = null;
```

Comment:
i don't think the existing behavior is that bad: basically if there are multiple h264's, we will prefer the first one we find, so i believe not even an error is really that necessary as the current behavior won't fail in that scenario (if it helps, i can add some more comments describing that behavior).  also, consider the two scenarios:
1) user is using jicofo: here there is only 1 h264 profile available
2) user is using their own signaling: here the user is in control of what the offer looks like, so they have some flexibility in what they offer already anyway.

not disagreeing that it's a good idea, i'm just not sure it feels urgent to worry about it now.  what do you think @paweldomas @saghul ?